src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets

```diff
@@ -312,29 +321,52 @@
     </PropertyGroup>
       
     <GenerateCurrentVersion SeedDate="$(VersionSeedDate)" OfficialBuildId="$(OfficialBuildId)" ComparisonDate="$(VersionComparisonDate)" Padding="$(VersionPadding)">
-      <Output PropertyName="GeneratedBuildNumberMajor" TaskParameter="GeneratedVersion" />
-      <Output PropertyName="GeneratedBuildNumberMinor" TaskParameter="GeneratedRevision" />
+      <Output PropertyName="BuildNumberMajor" TaskParameter="GeneratedVersion" />
+      <Output PropertyName="BuildNumberMinor" TaskParameter="GeneratedRevision" />
     </GenerateCurrentVersion>
-  </Target>
-
-  <Target Name="GetVersionSeedDate" Condition="'$(VersionSeedDate)'==''">
-    <ItemGroup>
-      <VersionTargetsFile Include="$(MSBuildThisFileFullPath)" />
-    </ItemGroup>
+    <PropertyGroup>
+      <VersionPadding/>
+      <VersionComparisonDate/>
+    </PropertyGroup>
       
-    <!-- Windows Exec command requires DOS escaping for the percent sign since it secretly runs the process in a shell instead of calling createprocess. -->
     <PropertyGroup>
-      <GitCommand Condition="'$(OsEnvironment)'=='Windows_NT'">git show -s --format=^%25%25cd --date=short HEAD</GitCommand>
-      <GitCommand Condition="'$(OsEnvironment)'!='Windows_NT'">git show -s --format=%25cd --date=short HEAD</GitCommand>
+      <VersionFileContent>
+        <![CDATA[<?xml version="1.0" encoding="utf-8"?>
+<!-- This is a generated file. $(VersionSourceComment) Seed Date is $(VersionSeedDate). -->
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <VersionPropsImported>true</VersionPropsImported>
+    <BuildNumberMajor Condition="'%24(BuildNumberMajor)' == ''">$(BuildNumberMajor)</BuildNumberMajor>
+    <BuildNumberMinor Condition="'%24(BuildNumberMinor)' == ''">$(BuildNumberMinor)</BuildNumberMinor>
+    <LatestCommit Condition="'%24(LatestCommit)' == ''">$(LatestCommit)</LatestCommit>
+    <BuiltByString Condition="'%24(BuiltByString)' == ''">$(BuiltByString)</BuiltByString>
+  </PropertyGroup>
+</Project>
+]]>
+      </VersionFileContent>
     </PropertyGroup>
+    
+    <!-- Since by default the file will get dropped at the obj dir, make sure that the dir is created already or else WriteLinesToFile will error. -->
+    <MakeDir Condition="!Exists('$(BuildVersionFilePath)')" Directories="$(BuildVersionFilePath)" />
```

Comment:
You don't need the condition, typically folks leave it out and the task has a fast path.  That's the pattern followed by common targets.